### ISSUE TITLE
postprocessing on MC cannot handle split by lumi

### ISSUE THREAD

**Issue by arizzi** [cms-nanoAOD/nanoAOD-tools#61]
Running on grid with split by lumi for MC results in duplicated counting information.
While splitting by file is an option, it doesn't allow to tune on number of events (EventsAwareLumiBased) that is instead useful.

**Comment by AndrewLevin** [comment]
Is someone working on a pull request for this? I can try to create one if not. This issue is basically the same as https://github.com/cms-nanoAOD/nanoAOD-tools/issues/28, right? There should be a histogram that is filled per event.

**Comment by arizzi** [comment]
well one can implement count histograms like we had in heppy or implement some logic in the postprocessor to properly update the run products..

**Comment by AndrewLevin** [comment]
But for samples with negative weights, I think using count histograms is basically the only option, and even this will not work when there is a cut string preselection.

**Comment by arizzi** [comment]
I think if you process by file there is no problem, the run products contains information taking into account the negatives too

**Comment by arizzi** [comment]
there is genEventSumw

**Comment by AndrewLevin** [comment]
yes, I agree, the problem is only when you process less than a full file

**Comment by arizzi** [comment]
well, if you want to process less than a full file then you should avoid the cut string and we can instead select in a first module in the chain that does the count booking and selection

**Comment by alefisico** [comment]
Hi all
I am actually having this problem. The jobs in the postprocessing step are really heavy for FileBased splitting in crab and I have been trying other splittings unsuccessfully. Does anybody have a suggestion how can I split the postprocessing jobs in other way that is not FileBased?
thanks!

**Comment by arizzi** [comment]
you can use count histograms instead of run products (i.e. you count yourself the weights)

btw if you have a post processing going slower than few tens of Hz (so hitting the limit of crab job duration) probably there is something bad in the rest of the configuration you run... which speed you achieve in your processing?

**Comment by alefisico** [comment]
Thanks @arizzi for the quick reply.
Although I am not sure if I follow your answer. What do you mean by "you can use count histograms instead of run products", which job splitting in crab should I use for this? 
About the second question: I have actually just send postprocessing jobs just caluclating JECs and btagging, but for some datasets in 2018, it seems that it is too heavy.

**Comment by arizzi** [comment]
you can use something like this
https://github.com/cms-nanoAOD/nanoAOD-tools/blob/master/python/postprocessing/modules/common/countHistogramsModule.py

to have the correct normalization of the number of events the job effectively run on. The run product counters should then be normalized to the value of the counting histogram (e.g. you can basically check the counting resulting in the Run tree and compute a SF from Nevents in histo/Run and use it for any Run product
